pkg/core/data/ipc/transactions/payload_decoded.go

```diff
@@ -193,3 +193,17 @@ func UnmarshalTransactionPayloadDecoded(r *bytes.Buffer, f *TransactionPayloadDe
 
 	return nil
 }
+
+// EstimatedGasSpent return the estimated amount of Gas that a transaction will
+// spend. For now it uses `t.Call` to determine if it's a normal transfer or it
+// is an intercontract call.
+// TODO: further improvement should take care of the contractAddress to compare
+// it against historical data.
+func (p *TransactionPayloadDecoded) EstimatedGasSpent() uint64 {
+	if p.Call == nil {
+		// This is a normal transfer transaction
+		return 250_000_000
+	}
+	// This is an inter-contract call
+	return 1_200_000_000
```

Comment:
Would suggest doing it the other way around.
The "normal" transaction is what should be in the bottom-most return.
The "non-normal" (inter-contract call) should be in the `if` block return.

```go
	if p.Call != nil {
		// This is an inter-contract call
		return 1_200_000_000
	}
	// Normal transfer transactions
	return 250_000_000
```

Reply:
In the future the "inter-contract call" branch will be longer (as per historical comparison)

So I prefer to leave it outer